### DISCUSSION TITLE
Corrosive effects using  openFast

### DISCUSSION THREAD

**Discussion by Krishnaraj18** [OpenFAST/openfast#1793] [Q&A]
how can we account for corrosion in OC4 jacket in openFast?

**Comment by jjonkman** [comment]
Dear @Krishnaraj18,

Can you clarify what specifically about corrosion you want to account for in OpenFAST?  Are you referring to changes in mass or stiffness?

Best regards,

**Reply by Krishnaraj18** [reply]
I have to do fragility analysis of OC4 jacket,for that I have to consider corrosion also.Usually in sacs software we will consider the corrosion by giving corrosion allowance.Is there any option in openfast to consider the corrosion effect of jacket?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

I'm not what SACS is specifically doing in this regard, but my understanding of corrosion allowance is a thickness of material added to allow for in-service corrosion.  Within OpenFAST, you can certainly set the thickness of each member.

Best regards

**Reply by Krishnaraj18** [reply]
i added some joint to divide the memeber in splash zones.I considered a depth of 4 to -2 as splash zone.And i divided the member in that depth by addidnf extra joints.Reduced the thickness of that member also.But while simulating it showing error like " **Error in file ".\NRELOffshrBsline5MW_OC4Jacket_SubDyn.dat": Reaction lines must consist of 7 numerical values, followed by an optional string. Problematic line:** ""Why it's coming like this?

**Reply by jjonkman** [reply]
Sounds like a formatting issue with your SubDyn input file.  As with any input file processing error, I would suggest enabling the `Echo` option to debug.

Best regards,

**Reply by Krishnaraj18** [reply]
But still facing the same issue. I am attaching the code that I edited along with this can you please check.
---- STRUCTURE JOINTS: joints connect structure members (~Hydrodyn Input File) --------
            64   NJoints     - Number of joints (-)
JointID          JointXss               JointYss               JointZss     JointType JointDirX  JointDirY JointDirZ JointStiff     ![Coordinates of Member joints in SS-Coordinate System][JointType={1:cantilever, 2:universal joint, 3:revolute joint, 4:spherical joint}]
  (-)               (m)                    (m)                    (m)         (-)        (-)        (-)       (-)     (Nm/rad) 
   1              6.00000                6.00000              -45.50000        1         0.0        0.0       0.0       0.0    
   2              6.00000                6.00000              -45.00000        1         0.0        0.0       0.0       0.0    
   3              5.96700                5.96700              -44.00100        1         0.0        0.0       0.0       0.0    
   4              5.93900                5.93900              -43.12700        1         0.0        0.0       0.0       0.0    
   5              5.33300                5.33300              -24.61400        1         0.0        0.0       0.0       0.0    
   6             -6.00000                6.00000              -45.50000        1         0.0        0.0       0.0       0.0    
   7             -6.00000                6.00000              -45.00000        1         0.0        0.0       0.0       0.0    
   8             -5.96700                5.96700              -44.00100        1         0.0        0.0       0.0       0.0    
   9             -5.93900                5.93900              -43.12700        1         0.0        0.0       0.0       0.0    
  10             -5.33300                5.33300              -24.61400        1         0.0        0.0       0.0       0.0    
  11             -6.00000               -6.00000              -45.50000        1         0.0        0.0       0.0       0.0    
  12             -6.00000               -6.00000              -45.00000        1         0.0        0.0       0.0       0.0    
  13             -5.96700               -5.96700              -44.00100        1         0.0        0.0       0.0       0.0    
  14             -5.93900               -5.93900              -43.12700        1         0.0        0.0       0.0       0.0    
  15             -5.33300               -5.33300              -24.61400        1         0.0        0.0       0.0       0.0    
  16              6.00000               -6.00000              -45.50000        1         0.0        0.0       0.0       0.0    
  17              6.00000               -6.00000              -45.00000        1         0.0        0.0       0.0       0.0    
  18              5.96700               -5.96700              -44.00100        1         0.0        0.0       0.0       0.0    
  19              5.93900               -5.93900              -43.12700        1         0.0        0.0       0.0       0.0    
  20              5.33300               -5.33300              -24.61400        1         0.0        0.0       0.0       0.0    
  21              4.82000                4.82000               -8.92200        1         0.0        0.0       0.0       0.0    
  22              4.38500                4.38500                4.37800        1         0.0        0.0       0.0       0.0    

  23              4.01600                4.01600               15.65100        1         0.0        0.0       0.0       0.0    
  24              4.00000                4.00000               16.15000        1         0.0        0.0       0.0       0.0    
  25             -4.82000                4.82000               -8.92200        1         0.0        0.0       0.0       0.0    
  26             -4.38500                4.38500                4.37800        1         0.0        0.0       0.0       0.0    
  27             -4.01600                4.01600               15.65100        1         0.0        0.0       0.0       0.0    
  28             -4.00000                4.00000               16.15000        1         0.0        0.0       0.0       0.0    
  29             -4.82000               -4.82000               -8.92200        1         0.0        0.0       0.0       0.0    
  30             -4.38500               -4.38500                4.37800        1         0.0        0.0       0.0       0.0    
  31             -4.01600               -4.01600               15.65100        1         0.0        0.0       0.0       0.0    
  32             -4.00000               -4.00000               16.15000        1         0.0        0.0       0.0       0.0    
  33              4.82000               -4.82000               -8.92200        1         0.0        0.0       0.0       0.0    
  34              4.38500               -4.38500                4.37800        1         0.0        0.0       0.0       0.0    
  35              4.01600               -4.01600               15.65100        1         0.0        0.0       0.0       0.0    
  36              4.00000               -4.00000               16.15000        1         0.0        0.0       0.0       0.0    
  37              5.62000                0.00000              -33.37300        1         0.0        0.0       0.0       0.0    
  38             -5.62000                0.00000              -33.37300        1         0.0        0.0       0.0       0.0    
  39              0.00000                5.62000              -33.37300        1         0.0        0.0       0.0       0.0    
  40              0.00000               -5.62000              -33.37300        1         0.0        0.0       0.0       0.0    
  41              5.06400                0.00000              -16.37100        1         0.0        0.0       0.0       0.0    
  42             -5.06400                0.00000              -16.37100        1         0.0        0.0       0.0       0.0    
  43              0.00000                5.06400              -16.37100        1         0.0        0.0       0.0       0.0    
  44              0.00000               -5.06400              -16.37100        1         0.0        0.0       0.0       0.0    
  45              4.59200                0.00000               -1.95800        1         0.0        0.0       0.0       0.0    
  46             -4.59200                0.00000               -1.95800        1         0.0        0.0       0.0       0.0    
  47              0.00000                4.59200               -1.95800        1         0.0        0.0       0.0       0.0    
  48              0.00000               -4.59200               -1.95800        1         0.0        0.0       0.0       0.0    
  49              4.19300                0.00000               10.26200        1         0.0        0.0       0.0       0.0    
  50             -4.19300                0.00000               10.26200        1         0.0        0.0       0.0       0.0    
  51              0.00000                4.19300               10.26200        1         0.0        0.0       0.0       0.0    
  52              0.00000               -4.19300               10.26200        1         0.0        0.0       0.0       0.0    
  53              4.00000                4.00000               20.15000        1         0.0        0.0       0.0       0.0    
  54             -4.00000                4.00000               20.15000        1         0.0        0.0       0.0       0.0    
  55              4.00000               -4.00000               20.15000        1         0.0        0.0       0.0       0.0    
  56             -4.00000               -4.00000               20.15000        1         0.0        0.0       0.0       0.0    
  57              6.00000               -6.00000              -49.50000        1         0.0        0.0       0.0       0.0    
  58              6.00000                6.00000              -49.50000        1         0.0        0.0       0.0       0.0    
  59             -6.00000               -6.00000              -49.50000        1         0.0        0.0       0.0       0.0    
  60             -6.00000                6.00000              -49.50000        1         0.0        0.0       0.0       0.0    
  61              4.59470                4.59470               -2.00000        1         0.0        0.0       0.0       0.0 
  62             -4.59470                4.59470               -2.00000        1         0.0        0.0       0.0       0.0 
  63             -4.59470                -4.59470               -2.00000         1         0.0        0.0       0.0       0.0 
  64              4.59470               -4.59470               -2.00000        1         0.0        0.0       0.0       0.0 
  65              6.00000               -6.00000              -50.00100        1         0.0        0.0       0.0       0.0    
  66              6.00000                6.00000              -50.00100        1         0.0        0.0       0.0       0.0    
  67             -6.00000                -6.00000              -50.00100         1         0.0        0.0       0.0       0.0    
  68             -6.00000                 6.00000              -50.00100         1         0.0        0.0       0.0       0.0 

------------------- BASE REACTION JOINTS: 1/0 for Locked/Free DOF @ each Reaction Node ---------------------
             4   NReact      - Number of Joints with reaction forces; be sure to remove all rigid motion DOFs of the structure  (else det([K])=[0])
RJointID   RctTDXss    RctTDYss    RctTDZss    RctRDXss    RctRDYss    RctRDZss    SSIfile [Global Coordinate System]
  (-)       (flag)      (flag)      (flag)      (flag)      (flag)      (flag)      (string)
  65           1           1           1           1           1           1        ""
  66           1           1           1           1           1           1        ""
  67           1           1           1           1           1           1        "" 
  68           1           1           1           1           1           1        ""
------- INTERFACE JOINTS: 1/0 for Locked (to the TP)/Free DOF @each Interface Joint (only Locked-to-TP implemented thus far (=rigid TP)) ---------
             8   NInterf     - Number of interface joints locked to the Transition Piece (TP):  be sure to remove all rigid motion dofs
IJointID   ItfTDXss    ItfTDYss    ItfTDZss    ItfRDXss    ItfRDYss    ItfRDZss
  (-)       (flag)      (flag)      (flag)      (flag)      (flag)      (flag)
  24           1           1           1           1           1           1
  28           1           1           1           1           1           1
  32           1           1           1           1           1           1
  36           1           1           1           1           1           1
  53           1           1           1           1           1           1
  54           1           1           1           1           1           1
  55           1           1           1           1           1           1
  56           1           1           1           1           1           1
----------------------------------- MEMBERS -------------------------------------------
           112   NMembers    - Number of members (-)
MemberID   MJointID1   MJointID2   MPropSetID1   MPropSetID2  MType   COSMID   ![MType={1:beam circ., 2:cable, 3:rigid, 4:beam arb.}. COMSID={-1:none}]
  (-)         (-)         (-)          (-)           (-)        (-)      (-)
   1           1           2            2             2          1        -1
   2           2           3            2             2          1        -1
   3           3           4            2             2          1        -1
   4           4           5            2             2          1        -1
   5           6           7            2             2          1        -1
   6           7           8            2             2          1        -1
   7           8           9            2             2          1        -1
   8           9          10            2             2          1        -1
   9          11          12            2             2          1        -1
  10          12          13            2             2          1        -1
  11          13          14            2             2          1        -1
  12          14          15            2             2          1        -1
  13          16          17            2             2          1        -1
  14          17          18            2             2          1        -1
  15          18          19            2             2          1        -1
  16          19          20            2             2          1        -1
  17           5          21            3             3          1        -1
  18          21          61            3             3          1        -1
  19          22          23            3             3          1        -1
  20          23          24            3             3          1        -1
  21          10          25            3             3          1        -1
  22          25          62            3             3          1        -1
  23          26          27            3             3          1        -1
  24          27          28            3             3          1        -1
  25          15          29            3             3          1        -1
  26          29          63            3             3          1        -1
  27          30          31            3             3          1        -1
  28          31          32            3             3          1        -1
  29          20          33            3             3          1        -1
  30          33          64            3             3          1        -1
  31          34          35            3             3          1        -1
  32          35          36            3             3          1        -1
  33           8           3            1             1          1        -1
  34          13           8            1             1          1        -1
  35          13          18            1             1          1        -1
  36          18           3            1             1          1        -1
  37           4          37            1             1          1        -1
  38          37          20            1             1          1        -1
  39          19          37            1             1          1        -1
  40          37           5            1             1          1        -1
  41           9          38            1             1          1        -1
  42          38          15            1             1          1        -1
  43          14          38            1             1          1        -1
  44          38          10            1             1          1        -1
  45           4          39            1             1          1        -1
  46          39          10            1             1          1        -1
  47           9          39            1             1          1        -1
  48          39           5            1             1          1        -1
  49          19          40            1             1          1        -1
  50          40          15            1             1          1        -1
  51          14          40            1             1          1        -1
  52          40          20            1             1          1        -1
  53           5          41            1             1          1        -1
  54          41          33            1             1          1        -1
  55          20          41            1             1          1        -1
  56          41          21            1             1          1        -1
  57          10          42            1             1          1        -1
  58          42          29            1             1          1        -1
  59          15          42            1             1          1        -1
  60          42          25            1             1          1        -1
  61           5          43            1             1          1        -1
  62          43          25            1             1          1        -1
  63          10          43            1             1          1        -1
  64          43          21            1             1          1        -1
  65          20          44            1             1          1        -1
  66          44          29            1             1          1        -1
  67          15          44            1             1          1        -1
  68          44          33            1             1          1        -1
  69          21          45            1             1          1        -1
  70          45          34            1             1          1        -1
  71          33          45            1             1          1        -1
  72          45          22            1             1          1        -1
  73          25          46            1             1          1        -1
  74          46          30            1             1          1        -1
  75          29          46            1             1          1        -1
  76          46          26            1             1          1        -1
  77          21          47            1             1          1        -1
  78          47          26            1             1          1        -1
  79          25          47            1             1          1        -1
  80          47          22            1             1          1        -1
  81          33          48            1             1          1        -1
  82          48          30            1             1          1        -1
  83          29          48            1             1          1        -1
  84          48          34            1             1          1        -1
  85          22          49            1             1          1        -1
  86          49          35            1             1          1        -1
  87          34          49            1             1          1        -1
  88          49          23            1             1          1        -1
  89          26          50            1             1          1        -1
  90          50          31            1             1          1        -1
  91          30          50            1             1          1        -1
  92          50          27            1             1          1        -1
  93          22          51            1             1          1        -1
  94          51          27            1             1          1        -1
  95          26          51            1             1          1        -1
  96          51          23            1             1          1        -1
  97          34          52            1             1          1        -1
  98          52          31            1             1          1        -1
  99          30          52            1             1          1        -1
 100          52          35            1             1          1        -1
 101          24          53            4             4          1        -1
 102          28          54            4             4          1        -1
 103          32          56            4             4          1        -1
 104          36          55            4             4          1        -1
 105          58           1            5             5          1        -1
 106          57          16            5             5          1        -1
 107          60           6            5             5          1        -1
 108          59          11            5             5          1        -1
 109          66          58            6             6          1        -1
 110          65          57            6             6          1        -1
 111          68          60            6             6          1        -1
 112          67          59            6             6          1        -1
 113          61          22            8             8          1        -1
 114          62          26            8             8          1        -1
 115          63          30            8             8          1        -1
 116          64         34            8             8          1        -1
------------------ CIRCULAR BEAM CROSS-SECTION PROPERTIES -----------------------------
             6   NPropSets   - Number of structurally unique cross-sections
PropSetID     YoungE          ShearG          MatDens          XsecD           XsecT
  (-)         (N/m2)          (N/m2)          (kg/m3)           (m)             (m)
   1        2.10000e+11     8.07690e+10       7850.00         0.800000        0.020000
   2        2.10000e+11     8.07690e+10       7850.00         1.200000        0.050000
   3        2.10000e+11     8.07690e+10       7850.00         1.200000        0.035000
   4        2.10000e+11     8.07690e+10       7850.00         1.200000        0.040000
   5        2.10000e+11     8.07690e+10       3339.12         2.082000        0.491000
   6        2.10000e+11     8.07690e+10       7850.00         2.082000        0.060000
   7        2.10000e+11     8.07690e+10       7850.00         0.800000        0.014000
   8        2.10000e+11     8.07690e+10       7850.00         1.200000        0.029000

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

If your issue is related to the formatting of your input file, posting the contents of your input file into this text editor will not be useful because the formatting may be changed by the editor.  If you want someone to help you, attach the input file and corresponding `Echo` file or a link to a repository where these can be downloaded.

Best regards,

**Reply by Krishnaraj18** [reply]
Ok sir .I am attaching fst ,subdyn and echo file along with this text,I edited the subdyn file previously for considering corrosion, please check is there any mistake. because while doing this simulation error  showing in subdyn file.
[echo.pdf](https://github.com/OpenFAST/openfast/files/12926874/echo.pdf)
[fst.pdf](https://github.com/OpenFAST/openfast/files/12926876/fst.pdf)
[subdyn.pdf](https://github.com/OpenFAST/openfast/files/12926877/subdyn.pdf)

**Reply by jjonkman** [reply]
I only see that you shared the `Echo` file for the OpenFAST primary (_.fst_) input file.  Can you also share the `Echo` associated with SubDyn?

That said, I see two blank lines in your Joint table, which will cause problems.  Does you model run correctly when you delete blank lines 50 and 97?

Best regards,

**Reply by Krishnaraj18** [reply]
I am attaching subDyn as well as it's echo file along with this,
[SubDyn echo.pdf](https://github.com/OpenFAST/openfast/files/13012426/SubDyn.echo.pdf)
[subdyn.pdf](https://github.com/OpenFAST/openfast/files/13012430/subdyn.pdf)

**Reply by jjonkman** [reply]
Both of your new attachments appear to be `Echo` files from SubDyn.  I assume that the previous SubDyn input file you've attached is correct.  Regardless, as I expected, it appears that the blank lines are causing SubDyn to have trouble interpreting the input file.  Also, I see that you've specified `NJoints` = 64, but you've actually specified 68 joints.  SubDyn believes you've entered `NReact` = 66, which is a value entered in the joint table.

Best regards,

**Reply by Krishnaraj18** [reply]
I simulated again after doing the corrections that you said, but it shows like
FAST_InitializeAll:SD_Init:SD_Input:Invalid numerical input for file
".\NRELOffshrBsline5MW_OC4Jacket_SubDyn.dat" occurred while trying to read NReact.

 OpenFAST encountered an error during module initialization.
 Simulation error level: FATAL ERROR

**Reply by jjonkman** [reply]
It still sounds like you have formatting issues with your SubDyn file.  I would suggest carefully comparing your SubDyn input file against the SubDyn `Echo` file to identify where the formatting issue is.

**Reply by Krishnaraj18** [reply]
While comparing the subdyn with it's echo file the code reading upto that joints.Still i am facing the issues,can you suggest me any other method to incorporate the corrosion effect in a water depth of -2 to 6 m?I want to reduce the thickness if member at the splash zone,how can i do that in other manner?And i also want to know that ,how we can define the splash zone depth in case of offshore wind turbine?also what is the thickness value we are going to take as corrosion allowance?
Another doubt is regarding PSI (Pile Soil interaction) ,where can i get PSI  file?,It is already coded or not?

**Reply by jjonkman** [reply]
To change the thickness of members around the splash zone, you should modify the structural members in SubDyn and hydrodynamic members in HydroDyn.  The errors you are facing sound like simple formatting issues.

The IEC 61400-3-1 and -3-2 design standards for offshore wind provide clear definitions of splash zone for fixed-bottom and floating offshore wind turbines.

By "PSI file", presumably you are referring to SubDyn's `SSIfile`; is that correct?  The `SSIfile` is supported by SubDyn, as documented here: https://openfast.readthedocs.io/en/main/source/user/subdyn/input_files.html#ssi-input-file.

Best regards,

**Reply by Krishnaraj18** [reply]
Sir i tried my maximum to find the correction in the Subdyn code i edited,i can't find the error in the code.In subDyn echo file upto joint there is no error found after that i edited the Nreaction thing only the joint name.Can you check what is the error in my code once more.

**Reply by Krishnaraj18** [reply]
Now my problem is fixed. The correction is nothing but i have to add that newly created joints in NReact part also. After editing that part it is running. And i checked the BM at mudline(TwrBsMxt-Side to side,TwrBsMyt-Fore-aft,TwrBsMzt-Yaw) of OC4 jacket for 30 m/s wind(Coupled) in the case of without adding corrosion and considering the corrosion also. From the results i ma getting increased side to side (TwrBsMxt)moment and Yaw moment (TwrBsMzt) in the second case(Considering the corrosion effects)but i found reduction in fore aft moment. I Is it correct or not? I have doubt that why it's coming like this. Can you please clarify it.

**Reply by Krishnaraj18** [reply]
How i will get nacelle acceleration ,transition piece displacement and tower top drift as output?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

I'm not sure I understand enough about what you changed to comment on how that would impact your results.

You can find the list of available outputs that can be included in the ElastoDyn and SubDyn `OutList`s here: https://openfast.readthedocs.io/en/main/_downloads/3f19498a5dc774461e022b671ff01ec6/OutListParameters.xlsx.

Best regards,

**Reply by Krishnaraj18** [reply]
How can I incorporate ROSCO controller in OpenFast, in the case of OC4 jacket?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

All of the NREL 5-MW baseline wind turbine models available in the OpenFAST r-test make use of the baseline controller for the NREL 5-MW turbine, not ROSCO.  But you could replace this baseline controller with a controller implemented in ROSCO (if you have ROSCO inputs set accordingly) by setting `DLL_FileName` and `DLL_InFile` within ServoDyn to point to the appropriate ROSCO dynamic library and ROSCO input file.

Best regards

**Reply by Krishnaraj18** [reply]
I  referred one of your journal "The Effect of Second-Order  Hydrodynamics on a Floating  Offshore Wind Turbine". In this journal the significant wave height and peak period value found only up to 50 m/s.  Can you please share the wave data(Hs,Tp)up to a  wind speed of 120 m/s.I am attaching the graph along with this.
![IMG-20231111-WA0001](https://github.com/OpenFAST/openfast/assets/68527519/88255cc1-0b27-4be1-839a-aef9fa64fc91)

**Reply by Krishnaraj18** [reply]
I have one more doubt,can I get  blade root displacement from openfast?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

I don't believe I have access to the data used to generate that Figure and certainly not for higher wind speeds.

The elastic deflection of the blade at the root is zero (because the blade is cantilevered to the hub).

Best regards,

**Reply by Krishnaraj18** [reply]
How can I draw fragility curve of OC4 jacket wind turbine using tower top displacement?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

Can you clarify what you mean?  Presumably you want to plot the probability of exceedance of something with respect to something else?

Best regards,

**Reply by Krishnaraj18** [reply]
Yes ,I want to plot  the probability of exceedance of tower top displacement Vs Wind velocity using incremental dynamic analysis. Do u have any idea how can I do. I have time series of tower top displacement, I want to draw find probability of exceedance from that value. How can I do that?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

I'm not really familiar with incremental dynamic analysis.

General post-processing tools such as MCrunch can calculate probability of exceedance of a given time series or set of time series.

Please note that your questions imply that you want to consider some sort of nonlinear push-over analysis, but you also mention that you are modeling the OC4 jacket in OpenFAST, so, presumably you are making use of the SubDyn module.  However, SubDyn is limited to linear elasticity.

Best regards,

**Reply by Krishnaraj18** [reply]
sir, I have one  doubt  regarding the natural frquency.How can I calculate natural frequency of the entire OC4 jacket wind turbine?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

You can use the full-system linearization functionality of OpenFAST, followed by Eigenanalysis, to determine the natural frequencies of land-based, fixed-bottom offshore, and floating offshore wind turbines.  This topic has been discussed many times in other posts.

Best regards,

**Reply by Krishnaraj18** [reply]
sir, I am doing dynamic analysis corresponding to indian ocean ,so instead of JONSWAP i have to give PM spectrum.But in OpenFast JONSWAP/PM is coming as an combined option in waveMod (   WaveMod  - Incident wave kinematics model {0: none=still water, 1: regular (periodic), 1P#: regular with user-specified phase, 2: JONSWAP/Pierson-Moskowitz spectrum (irregular)).So if I am giving PM what are the other input parameter I have to change. Because its coming as same option 2.If i give WaveMod 2 means which spectrum it will read?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

The JONSWAP spectrum is a simple modification of the Pierson-Moskowitz spectrum, making the spectrum more "peaky" for storm conditions.  To use the Pierson-Moskowitz spectrum, set `WaveMod` = 2 with `WavePkShp` = 1.0 in HydroDyn.

Best regards

**Comment by Krishnaraj18** [comment]
Dear Jason,
Is it possible to edit the member properties (thickness) of the members in OC4 jacket model using subdyn and will it cause any change in response of the structure. Also can we reduce the thickness of OC3 monpile in subdyn. I've tried to do the above mentioned tasks but the results are not changing. Do I only need to change the code in subdyn.dat file or should i do anything else?

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

Yes, you can modify the member properties (including thickness) within the SubDyn input file and I would expect this to change the results (loads, motions, though, the change in results may be minor for minor changes in member thickness).

Best regards,

**Comment by Krishnaraj18** [comment]
Dear Jason,
In OPenFAST I had edited the member properties (thickness) in the "subdyn.dat" file of the OC3 monopile supported OWT, but the corresponding reflections are shown in the subdyn summary file. So the loads and deflections are not changing, but  instead I am getting the exact same value.( I reduced the member thickness from 6cm to 5cm).

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

I'm sorry, but I don't know what you mean when you say that, "the corresponding reflections are shown in the subdyn summary file."  Are you saying that the thickness changes are seen in the SubDyn summary file in terms of changes to the masses, eigenfrequencies, etc., but are not seen in the SubDyn output file in terms of loads and deflections?

Best regards,

**Reply by Krishnaraj18** [reply]
Dear Jason, 
I am extremely sorry for the typing error. What I really meant was the "the corresponding reflections are not shown in summary file".

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

So, you are not seeing any change to any SubDyn output when you change the member thickness?  I would not expect this.  Are you saving the file after making the input file change?  Are you actually using the cross-sectional properties in a member for the `PropSetID` you are changing?

Best regards,

**Comment by Krishnaraj18** [comment]
Dear Jason,
Now when I checked the summary file is not generating even after giving "True" for SumPrint. I am attaching my sundyn.dat file along with this .Please say what to do.

[5MW_OC3Mnpl_DLL_WTurb_WavesIrr.txt](https://github.com/OpenFAST/openfast/files/13725609/5MW_OC3Mnpl_DLL_WTurb_WavesIrr.txt)

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

Can you confirm that `SumPrint` is being interpreted correctly by reviewing the echo (_.ech_) file?

Does changing `SumPrint` from 1 to True solve the issue?

Best regards,

**Comment by Krishnaraj18** [comment]
Dear Jason,
I am currently working on the 5MW NREL offshore wind turbine supported over OC3 monopile. I've been trying to change the thickness of the monopile from 6cm up to 10cm and planning to conduct fatigue analysis. For that I edited the thickness in "subdyn.dat" file to 10cm, but unfortunately the results are showing the same as for 6cm thickness. Can you explain me the reason for this paradox. I am attaching the corresponding files with this reply, can you please look into it.
[Output_file_6cm_thickness.txt](https://github.com/OpenFAST/openfast/files/13958938/Output_file_6cm_thickness.txt)
[Output_file_10cm_thickness.txt](https://github.com/OpenFAST/openfast/files/13958941/Output_file_10cm_thickness.txt)
[SubDyn.dat _For_6cm_thickness.txt](https://github.com/OpenFAST/openfast/files/13958942/SubDyn.dat._For_6cm_thickness.txt)
[SubDyn.dat _For_10cm_thickness.txt](https://github.com/OpenFAST/openfast/files/13958943/SubDyn.dat._For_10cm_thickness.txt)
[SubDyn.sum _For_6cm_thickness.txt](https://github.com/OpenFAST/openfast/files/13958944/SubDyn.sum._For_6cm_thickness.txt)
[SubDyn.sum _For_10cm_thickness.txt](https://github.com/OpenFAST/openfast/files/13958946/SubDyn.sum._For_10cm_thickness.txt)

**Reply by jjonkman** [reply]
Dear @Krishnaraj18,

I'm not sure I understand your question.  I see a lot of differences when I compare the SubDyn summary files and OpenFAST output files for your two SubDyn input files.

Please note that I see you changed more than just the thickness between these two SubDyn input files.  In the 6cm file, you've selected `MPropSetID1` = `MPropSetID2` = 1 for all members, whereas in the 10cm file, you've selected `MPropSetID1` = `MPropSetID2` = 3 for all members.

Best regards,
